internal/pkg/template/templates/workloads/partials/cf/nlb.yml

```
@@ -10,16 +10,48 @@ PublicNetworkLoadBalancer:
         - Fn::ImportValue:
             !Sub '${AppName}-${EnvName}-PublicSubnets'
     Type: network
+    SecurityGroups:
+      - !GetAtt NetworkLoadBalancerSecurityGroup.GroupId
+NetworkLoadBalancerSecurityGroup:
+  Metadata:
+    'aws:copilot:description': 'A security group for your network load balancer allowing public traffic'
+  Type: AWS::EC2::SecurityGroup
+  Properties:
+    GroupDescription: HTTP access to the public facing load balancer
```

Since it's not necessarily HTTP -
```suggestion
    GroupDescription: allow traffic to the public facing load balancer on listener ports.
```